docs/pages/reference/access-controls/roles.mdx

```diff
@@ -388,7 +390,32 @@ have a more permissive role, like the preset `auditor` role, which allows access
 to all events, sessions, and session recordings.
 </Admonition>
 
-Role for restricted access to session recordings:
+### Understanding the `where` clause
+
+The `where` clause acts as a condition that must evaluate to true for access to be
+granted. It uses predicate expressions that can reference:
+
+- Attributes of the authenticated user (name, roles, traits)
+- Metadata about the session recording (participants, proto, creator, etc.)
+- Resource-specific attributes (server details, database info, etc.)
+
+Basic syntax example:
+
+```
+where: contains(session.participants, user.metadata.name)
+```
+
+This expression evaluates to true only when the current user is found in the session's
+participants list.
+
+
+### Common Access Control Recipes
+
+Here are practical examples of how to use `where` expressions to implement common access control patterns:
+
+1. Restrict to Session Participants
```

Comment:
These examples are great, can we turn these titles into proper markdown headers (`####`) so they can be linked to?